tests/e2e/kcs/test_noobaa_rebuild.py

```diff
@@ -0,0 +1,178 @@
+        4. Patch existing backingstores and bucketclasses to remove finalizer
+        5. Delete the backingstores/bucketclass.
+        6. Delete the noobaa secrets.
+        7. Restart noobaa-operator by setting the replicas back to 1.
+        8. Monitor the pods in openshift-storage for noobaa pods to be Running.
+
+        """
+        dep_ocp = OCP(
+            kind=constants.DEPLOYMENT, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        state_ocp = OCP(
+            kind=constants.STATEFULSET, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        noobaa_pvc_obj = get_pvc_objs(pvc_names=["db-noobaa-db-pg-0"])
+
+        # Scale down noobaa operator
+        logger.info("Scaling down noobaa-operator to replica: 0")
+        dep_ocp.exec_oc_cmd("scale deployment noobaa-operator --replicas=0")
+
+        # Delete noobaa deployments and statefulsets
+        logger.info("Deleting noobaa deployments and statefulsets")
+        dep_ocp.delete(resource_name="noobaa-endpoint")
+        state_ocp.delete(resource_name="noobaa-db-pg")
+        state_ocp.delete(resource_name="noobaa-core")
+
+        # Delete noobaa-db pvc
+        pvc_obj = OCP(
+            kind=constants.PVC, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        logger.info("Deleting noobaa db pvc")
+        pvc_obj.delete(resource_name="db-noobaa-db-pg-0", wait=True)
+
+        # Patch and delete existing backingstores
+        params = '{"metadata": {"finalizers":null}}'
+        bs_obj = OCP(
+            kind=constants.BACKINGSTORE, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        for bs in bs_obj.get()["items"]:
+            assert bs_obj.patch(
+                resource_name=bs["metadata"]["name"],
+                params=params,
+                format_type="merge",
+            ), "Failed to change the parameter in backingstore"
+            logger.info(f"Deleting backingstore: {bs['metadata']['name']}")
+            bs_obj.delete(resource_name=bs["metadata"]["name"])
+
+        # Patch and delete existing bucketclass
+        bc_obj = OCP(
+            kind=constants.BUCKETCLASS, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        for bc in bc_obj.get()["items"]:
+            assert bc_obj.patch(
+                resource_name=bc["metadata"]["name"],
+                params=params,
+                format_type="merge",
+            ), "Failed to change the parameter in bucketclass"
+            logger.info(f"Deleting bucketclass: {bc['metadata']['name']}")
+            bc_obj.delete(resource_name=bc["metadata"]["name"])
+
+        # Delete noobaa secrets
+        logger.info("Deleting noobaa related secrets")
+        dep_ocp.exec_oc_cmd(
+            "delete secrets noobaa-admin noobaa-endpoints noobaa-operator noobaa-server"
+        )
+
+        # Scale back noobaa-operator deployment
+        logger.info("Scaling back noobaa-operator to replica: 0")
```

Comment:
we are scaling up back, it should be replica:1

Reply:
changed